### ISSUE TITLE
404 error while requesting /wp-jason/

### ISSUE THREAD

**Issue by kdpuvvadi** [WP-API/WP-API#1509]
i've permalink structure as /post-name/ and and plugin is activated, when i request /wp-json/ from rest console or from my IDE(Visual studio) i'm getting 404 error and i thought some thing wrong with VS and tested in browser and it is redirecting to 404 page. website is running on Ubuntu 15, with apache 2.4, basic lamp setup on Azure.

And super cache plugin is enabled on wordpress setup.
How can i possible solve this anyway?

**Comment by rachelbaker** [comment]
@kdpuvvadi Try using `/wp-json/ ` instead.  You appear to have an errant `a` in the path you are requesting.

**Comment by kdpuvvadi** [comment]
it was spell mistake, i'm using /wp-json/ only

**Comment by joehoyle** [comment]
@kdpuvvadi are you able to access the API at `/?rest_route=/` ? (for example, demo.wp-api.org/?rest_route=/)

**Comment by kdpuvvadi** [comment]
yes and it is showing mysite.com/wp-json but is giving 404 error

**Comment by kdpuvvadi** [comment]
this is the response i'm getting 

[{"code":"rest_no_route","message":"No route was found matching the URL and request method","data":{"status":404}}]

**Comment by joehoyle** [comment]
@kdpuvvadi what version of the plugin are you using?

**Comment by kdpuvvadi** [comment]
1.2.3

**Comment by pateljj04** [comment]
I am having a similar issue, I can hit ?json_route=/posts&type=page and works fine, but using /wp-json/ shows 404. The routes seem to be setup - I have pretty links on and i have flushed rewrites. 
^wp-json/?$	index.php?json_route=/	other
^wp-json(.*)?	index.php?json_route=$matches[1]

**Comment by kdpuvvadi** [comment]
After Deactivating and reacting the plugin i'm getting error 500 in return,

**Comment by eliorivero** [comment]
Hi, can you verify that you're also adding the namespace in the request URL? it should be something like
`
http://yoursite.tld/wp-json/wp/v2/posts/
`
this is one of the routes that the plugin registers, the default namespace is wp/v2.

Note that if you're registering your endpoint, it's strongly suggested that you add your own namespace.

**Comment by pateljj04** [comment]
I think this issue is on the stable version not V2.

**Comment by rahulchaturvedie** [comment]
I just installed v2 and the issue is I am able to access http://example.com/?rest_route=/wp/v2/posts/ but not with http://example.com/wp-json/wp/v2/posts/ on default permalink

Now if I set the permalink to post name then also I am able to access http://example.com/?rest_route=/wp/v2/posts/ but not with http://example.com/wp-json/wp/v2/posts/ also changing to post name sends my regulart pages to 404 http://example.com/hello-world goes to 404

So for now I am setting it to default permalink and using http://example.com/?p=1 and http://example.com/?rest_route=/wp/v2/posts/

**Comment by kdpuvvadi** [comment]
This was fixed after upgrading to version 2 beta 7.

**Comment by Zeladgolan** [comment]
Just upgrade the wp plugin to latest, This will be fixed.

**Comment by hyyan** [comment]
I am having the same problem with beta13 , any idea ?

**Comment by hyyan** [comment]
Sorry , it is the server configuration fault , the rewrite module was not enabled

**Comment by zzimbler** [comment]
Just an FYI, currently running v2 of the plugin and Wordpress version 4.5.2 via MAMP on a mac and still get the 404 error. You can still access it at the ?rest_route url. Permalinks are disabled. This isn't really a big deal since we can access it, but it was a pain in the butt to initially figure this out

**Comment by TheDiamondYT1** [comment]
You need permalinks enabled

**Comment by edu-hub1107** [comment]
Its very simple, the namespace in the request URL must be like this -
http://domain.com/wp-json/posts/ 

I hope it works....

**Comment by engr-zeeshan** [comment]
"The [domain name] page isn't working" error when I use
http://mydomain/wp-json/wp/v2/posts

Permalinks are enabled

**Comment by ohohohoo** [comment]
So after upgrading into WP4.7 on my test machine (with wp-rest-api plugin still activated and later deactivated) I'm having this issue on my end too. I received JSON rest_no_route when accessing /wp-json/ directly, but when I tried to access it via ?rest_route=/ ... it works just fine.

Accessing custom route via /wp-json/my/own/custom/route is good. Accessing /wp-json/wp/v2/ also okay and it display all the namespace. But not /wp-json/.

Back then on WP4.6 with 2.15 Beta rest api plugin I never experienced this issue. This issue is appear after I upgraded to WP4.7.

Anyone has any idea on what could cause this?
